lib/train/platforms/detect/helpers/os_common.rb

```diff
@@ -79,6 +79,11 @@ def cisco_show_version
         return @cache[:cisco] = { version: m[2], model: m[1], type: 'ios-xe' }
       end
 
+      m = res.match(/Cisco IOS Software \[([^,]+)\], Virtual XE Software [^,]+, Version (\d+\.\d+\.\d+[A-Z]*)/)
```

Comment:
I can't decide if this is different enough that I wouldn't just modify the other regex for Cisco IOS XE. :thinking:

I think this is fine though. Maybe add a comment to show this is Virtual XE?

Reply:
Yeah, I also couldn't decide. I opted for a separate case for it since it's much easier to combine regexes than to take them apart. I'll add in comments to indicate why it's separate. 👍